pkg/pgmodel/ingestor/trace/writer.go

```diff
@@ -328,7 +328,7 @@ func (t *traceWriterImpl) InsertTraces(ctx context.Context, traces pdata.Traces)
 
 	code = "2xx"
 	metrics.IngestorInsertDuration.With(prometheus.Labels{"type": "trace", "subsystem": "", "kind": "span"}).Observe(time.Since(start).Seconds())
-	metrics.IngestorMaxSentTimestamp.With(traceLabel).Set(float64(maxEndTimestamp))
+	metrics.IngestorMaxSentTimestamp.With(traceLabel).Set(float64(maxEndTimestamp * 1e3))
```

Comment:
Isn't this in microseconds and we want milliseconds?

I suggest using `time.Time` for `maxEndTimestamp` and avoid these magic conversions.